web/src/stream_pill.ts

```diff
@@ -60,12 +60,28 @@ export function get_stream_name_from_item(item: StreamPill): string {
     return stream.name;
 }
 
-export function get_user_ids(pill_widget: StreamPillWidget | CombinedPillContainer): number[] {
-    let user_ids = pill_widget
-        .items()
-        .flatMap((item) =>
-            item.type === "stream" ? peer_data.get_subscribers(item.stream_id) : [],
-        );
+export async function get_user_ids(
+    pill_widget: StreamPillWidget | CombinedPillContainer,
+): Promise<number[]> {
+    const stream_ids = get_stream_ids(pill_widget);
+    const results = await Promise.all(
+        stream_ids.map(async (stream_id) => peer_data.get_all_subscribers(stream_id, false)),
+    );
+
+    const current_stream_ids_in_widget = get_stream_ids(pill_widget);
+    let user_ids: number[] = [];
+    for (const [index, stream_id] of stream_ids.entries()) {
+        const subscribers = results[index]!;
+        // (1) `null` means the request failed. We don't want to keep retrying
+        // since awaiting that will delay this function's return of other
+        // channel's data.
```

Comment:
OK but what's the user-facing outcome in that case --- I think we don't include any subscribers from this channel in the code below. Does the UI indicate that? 

I think it might be better to keep retrying from a UX standpoint, but haven't tested.

Reply:
Started a conversation about both if we want to retry and where/what the error message should be: [#frontend > pills adding channel subscribers, with partial data  #34715 @ 💬](https://chat.zulip.org/#narrow/channel/6-frontend/topic/pills.20adding.20channel.20subscribers.2C.20with.20partial.20data.20.20.2334715/near/2200945)

Reply:
As discussed on CZO, I'll make this do retries

Reply:
([diff here](https://github.com/zulip/zulip/compare/2abc8d8a6f3edfe8f453b423e765131af8252e36..193cea24805f038921703c6d769508d610b7cc40))